### PR TITLE
Integrate krnlmon into infrap4d instead of stratum

### PR DESCRIPTION
**Pull request by nupuruttarwar**
Earlier, krnlmon was tightly coupled into startum and krnlmon_init was called directly from stratum after P4 servers are in listening state. Since this is not a scalable solution, this PR removes direct integration of krnlmon from stratum. Instead, infrap4d spawns 3 threads -
  - krnlmon init
  - krnlmon_shutdown
  - Stratum main. Once P4RT servers are initialized as part of stratum main,it will signal krnlmon_init thread to resume execution.
Similarly, once P4RT servers are stopped, it will signal krnlmon_shutdown thread to resume execution

Signed-off-by: Nupur Uttarwar <nupur.uttarwar@intel.com>

### CODE REVIEW

**Comment by nupuruttarwar** [comment]
Keeping it as draft PR until stratum and krnlmon PR's are merged. Once they are merged, I will modify this PR to update the reference point for these submodules

File: infrap4d/CMakeLists.txt [hunk 1]
```diff
@@ -5,28 +5,37 @@ project(infrap4d VERSION 0.1.0 LANGUAGES C CXX)
 set(STRATUM_TDI_BIN_DIR ${STRATUM_SOURCE_DIR}/stratum/hal/bin/tdi)
 set(STRATUM_BFRT_BIN_DIR ${STRATUM_SOURCE_DIR}/stratum/hal/bin/barefoot)
 
-if(DPDK_TARGET)
+if(WITH_KRNLMON)
+    add_executable(infrap4d infrap4d_main.cc)
```
Comment (ffoulkes) [inline infrap4d/CMakeLists.txt:9]: Where is `infrap4d_main.cc`? I don't see it in `infrap4d`, and it's not part of this commit.

**Review by ffoulkes — CHANGES_REQUESTED** [review]
